modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -137,4 +139,102 @@
      * @return The current value
      */
     T getValue();
+
+    /**
+     * Creates an {@code ObservableValue} that holds the result of applying a
+     * mapping on this {@code ObservableValue}'s value. The result is updated
+     * when this {@code ObservableValue}'s value changes. If this value is
```

Comment:
I read this comment after what I wrote about `flatMap`, so mstr2 also had the idea of "More precisely", which is good :)

I would suggested something similar to what I did there:

```
Creates a new {@code ObservableValue} that holds the value supplied by the given mapping function. The result
is updated when this {@code ObservableValue} changes.
If this value is {@code null}...
More precisely, the created {@code ObservableValue} holds the result of applying a mapping on this
{@code ObservableValue}'s value.
```

Same comments about `@return` and `@throws` NPE as I had for `flatMap`.

`orElse` will also becomes something like

```
Creates a new {@code ObservableValue} that holds this value, or the given value if it is {@code null}. The
result is updated when this {@code ObservableValue} changes.
More precisely, the created {@code ObservableValue} holds this {@code ObservableValue}'s value, or
the given value if it is {@code null}.
```

Also not sure if the "More precisely" description is needed here.

Reply:
I think I've arrived at the conclusing that the "more precisely" description is basically rehashing the semantics of `ObservableValue` over and over again.
It should be clear for a reader who is familiar with ObservableValue that we're not suggesting that somehow the given ObservableValue _instance_ is stored in the new ObservableValue.

Reply:
@nlisker @mstr2 I've done another pass, using all the suggestions, and also by looking closely at how things are worded in the documentation for Optional and Stream.  This resulted in a few additional changes.  I've left out the "more precisely" parts as I think we all agree it adds little extra value.  Please have another look.

Reply:
I've changed the wording also to `Returns an` instead of `Creates an` or `Creates a new` -- I think we don't need to guarantee a new instance is always returned, just an instance that works as described (potentially cached or re-used).  Stream and Optional word this similar.